### PR TITLE
vcpu_misc: Add case for max cpu mode

### PR DESCRIPTION
**Pull request by dzhengfy**
Case ID: RHEL-242822
This case is to test maximum cpu mode.

Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

**Comment by dzhengfy** [comment]
libvirt-7.4.0-1.el9.x86_64
```
# avocado run --vt-type libvirt --vt-machine-type q35 vcpu_misc..mode_max
WARNING:root:No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
WARNING:root:No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : 5308ef6a0a20986a72002d16633d0824e4a0c3c3
JOB LOG    : /root/avocado/job-results/job-2021-06-22T06.36-5308ef6/job.log
 (1/1) type_specific.io-github-autotest-libvirt.vcpu_misc.positive_test.mode_max: PASS (46.66 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 47.18 s
# avocado run --vt-type libvirt --vt-machine-type q35 vcpu_misc
JOB ID     : aa0f938b970454354a492ff84283fb257dada7cf
JOB LOG    : /root/avocado/job-results/job-2021-06-22T06.38-aa0f938/job.log
 (1/6) type_specific.io-github-autotest-libvirt.vcpu_misc.positive_test.snapshot: PASS (6.17 s)
 (2/6) type_specific.io-github-autotest-libvirt.vcpu_misc.positive_test.vendor_id: PASS (43.30 s)
 (3/6) type_specific.io-github-autotest-libvirt.vcpu_misc.positive_test.with_topology.vcpus_max_equal_topology: PASS (45.53 s)
 (4/6) type_specific.io-github-autotest-libvirt.vcpu_misc.positive_test.mode_max: PASS (41.99 s)
 (5/6) type_specific.io-github-autotest-libvirt.vcpu_misc.negative_test.with_topology.vcpus_max_exceed_topology: PASS (25.59 s)
 (6/6) type_specific.io-github-autotest-libvirt.vcpu_misc.negative_test.with_topology.vcpus_max_less_than_topology: PASS (25.36 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 188.59 s

```

File: libvirt/tests/cfg/cpu/vcpu_misc.cfg [hunk 1]
```diff
@@ -21,6 +21,10 @@
                     variants:
                         - vcpus_max_equal_topology:
                             virsh_edit_cmd = ":%s #[0-9]*</vcpu>#8</vcpu>"
+                - mode_max:
+                    func_supported_since_libvirt_ver = (7, 3, 0)
+                    cpu_mode = 'maximum'
```
Comment (Yingshun) [inline libvirt/tests/cfg/cpu/vcpu_misc.cfg:26]: Can you create a new file for this test? It's about cpu_mode, so I think we shouldn't put it in vcpu_misc. What do you think?

Comment (dzhengfy) [reply libvirt/tests/cfg/cpu/vcpu_misc.cfg:26]: From my POV, I did not see any batch test for cpu mode. I also did not see more cases in backlog for cpu mode test, Usually we only test with 'host-model'. This misc file is very simple and easy to be reuse for this test. So currently I prefer to reuse it. If more tests for cpu mode comes in future, I can consider to reorg them in new file.

Comment (Yingshun) [reply libvirt/tests/cfg/cpu/vcpu_misc.cfg:26]: Okay, got it.

**Review by Yingshun — APPROVED** [review]

**Review by chloerh — APPROVED** [review]

**Pull request merged**